R/travis-auth.R

```diff
@@ -1,75 +1,103 @@
-auth_travis_ <- function(gh_token = NULL) {
+#' Authenticate to Travis
+#' @description
+#'   Authenticates to Travis and asks to create an API key if none exists.
+#' @template endpoint
+#' @export
+#'
+auth_travis <- function(endpoint = ".org") {
+  yml <- tryCatch(
+    {
+      readLines("~/.travis/config.yml")
+    },
+    warning = function(cond) {
```

Comment:
Catching all warnings is very brittle.

Reply:
Do you know how to do this in a more elegant way?

Reply:
I'd suggest to have the user set up keys up front, and fail promptly if the keys aren't found. No auto-magic here, because debugging any problems will be a nightmare.